third_party/tsl/tsl/cuda/cublas_stub.cc

```diff
@@ -26,43 +27,221 @@ limitations under the License.
+      "cublasScopy",
+      "cublasDcopy",
+      "cublasCcopy",
+      "cublasZcopy",
+      "cublasSswap",
+      "cublasDswap",
+      "cublasCswap",
+      "cublasZswap",
+      "cublasIsamax",
+      "cublasIdamax",
+      "cublasIcamax",
+      "cublasIzamax",
+      "cublasIsamin",
+      "cublasIdamin",
+      "cublasIcamin",
+      "cublasIzamin",
+      "cublasSasum",
+      "cublasDasum",
+      "cublasScasum",
+      "cublasDzasum",
+      "cublasSrot",
+      "cublasDrot",
+      "cublasCrot",
+      "cublasZrot",
+      "cublasCsrot",
+      "cublasZdrot",
+      "cublasSrotg",
+      "cublasDrotg",
+      "cublasCrotg",
+      "cublasZrotg",
+      "cublasSrotm",
+      "cublasDrotm",
+      "cublasSrotmg",
+      "cublasDrotmg",
+      "cublasSgemv",
+      "cublasDgemv",
+      "cublasCgemv",
+      "cublasZgemv",
+      "cublasSgbmv",
+      "cublasDgbmv",
+      "cublasCgbmv",
+      "cublasZgbmv",
+      "cublasStrmv",
+      "cublasDtrmv",
+      "cublasCtrmv",
+      "cublasZtrmv",
+      "cublasStbmv",
+      "cublasDtbmv",
+      "cublasCtbmv",
+      "cublasZtbmv",
+      "cublasStpmv",
+      "cublasDtpmv",
+      "cublasCtpmv",
+      "cublasZtpmv",
+      "cublasStrsv",
+      "cublasDtrsv",
+      "cublasCtrsv",
+      "cublasZtrsv",
+      "cublasStpsv",
+      "cublasDtpsv",
+      "cublasCtpsv",
+      "cublasZtpsv",
+      "cublasStbsv",
+      "cublasDtbsv",
+      "cublasCtbsv",
+      "cublasZtbsv",
+      "cublasSsymv",
+      "cublasDsymv",
+      "cublasChemv",
+      "cublasZhemv",
+      "cublasSsbmv",
+      "cublasDsbmv",
+      "cublasChbmv",
+      "cublasZhbmv",
+      "cublasSspmv",
+      "cublasDspmv",
+      "cublasChpmv",
+      "cublasZhpmv",
+      "cublasSger",
+      "cublasDger",
+      "cublasCgeru",
+      "cublasCgerc",
+      "cublasZgeru",
+      "cublasZgerc",
+      "cublasSsyr",
+      "cublasDsyr",
+      "cublasCher",
+      "cublasZher",
+      "cublasSspr",
+      "cublasDspr",
+      "cublasChpr",
+      "cublasZhpr",
+      "cublasSsyr2",
+      "cublasDsyr2",
+      "cublasCher2",
+      "cublasZher2",
+      "cublasSspr2",
+      "cublasDspr2",
+      "cublasChpr2",
+      "cublasZhpr2",
+      "cublasSgemm",
+      "cublasDgemm",
+      "cublasCgemm",
+      "cublasZgemm",
+      "cublasSsyrk",
+      "cublasDsyrk",
+      "cublasCsyrk",
+      "cublasZsyrk",
+      "cublasCherk",
+      "cublasZherk",
+      "cublasSsyr2k",
+      "cublasDsyr2k",
+      "cublasCsyr2k",
+      "cublasZsyr2k",
+      "cublasCher2k",
+      "cublasZher2k",
+      "cublasSsymm",
+      "cublasDsymm",
+      "cublasCsymm",
+      "cublasZsymm",
+      "cublasChemm",
+      "cublasZhemm",
+      "cublasStrsm",
+      "cublasDtrsm",
+      "cublasCtrsm",
+      "cublasZtrsm",
+      "cublasStrmm",
+      "cublasDtrmm",
+      "cublasCtrmm",
+      "cublasZtrmm",
+  };
+  return *syms;
+}
+
+} // namespace
+
+extern "C" {
+
+static void CublasLogFatalSymbolNotFound(const char *symbol_name) {
```

Comment:
Out of curiosity: why can't we use the `CublasGetSymbolNotFoundError` function in all cases?

Reply:
NVIDIA's APIs don't all return a status type. e.g., consider a symbol like:

```
CUBLASAPI size_t CUBLASWINAPI cublasGetCudartVersion(void);
```

It's probably better that we `LOG(FATAL)` than return a bogus value.

(I doubt this will matter in practice: the first thing you're going to do almost certainly is call `cublasCreate`, which does return a status.)

And happily almost all of NVIDIA's APIs *do* return a status.